src/components/auth/Login.jsx

```diff
@@ -1,14 +1,93 @@
+import { AuthProvider, useAuth } from "../../AuthProvider";
+import { Link, useNavigate } from "react-router-dom";
+import { useEffect, useState } from "react";
+
 import FormInput from "../common/FormInput";
+import { login } from "../../util/fetchData"
+
 const Login = () => {
+  const [formData, setFormData] = useState({ email: "", password: "" });
+  const [formErrors, setFormErrors] = useState({});
+  const { isLoggedIn, loginAction } = useAuth();
+  const navigate = useNavigate();
+
+  const handleChange = (e) => {
+    setFormData({ ...formData, [e.target.name]: e.target.value });
+    setFormErrors({ ...formErrors, [e.target.name]: '', form: '' });
+  }
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    if (!formData.email || !formData.password) {
+      setFormErrors({
+        ...formErrors,
+        email: !formData.email ? "Email is required" : null,
+        password: !formData.password ? "Password is required" : null,
+      });
+      return;
+    }
+
+    try {
+      const result = await login(formData);
+      if (result.status === 200) {
+        let userData = result.data.user;
+        loginAction(true, userData);
+        navigate("/dashboard");
+      }
+    } catch (error) {
+      setFormErrors({
+        ...formErrors,
+        form: error.message,
+      });
+    }
+  };
+
+  useEffect(() => {
+    if (isLoggedIn) {
+      navigate("/dashboard");
+    }
+  }, [isLoggedIn, navigate]);
+
   return (
-    <>
-      <h1>Login Form</h1>
-      {/* testing how the inputs looks like */}
-      <form>
-        <FormInput id="email" type="email" placeholder=" ">Email address</FormInput>
-        <FormInput id="password" type="password" placeholder=" ">Password</FormInput>
+    <div className="flex flex-col gap-5 mt-16 items-center">
+      <h1 className="font-spartan text-3xl">Log In</h1>
+      <form className=" w-2/3 md:w-1/2 lg:w-1/3 xl:w-3/12" onSubmit={handleSubmit}>
+        <p className="font-spartan mb-4">Don't have an account? <span><Link to="/register" className="underline">Sign Up</Link></span></p>
+        {/* without passing the empty string in the placeholder attribute the floating label doesn't go back into the input */}
+        {<p className="text-red text-sm font-spartan">{formErrors.email}</p>}
+        {formErrors.form === "Invalid email" 
```

Comment:
When credentials are invalid it's generally good practice to give a more generic reason for why validation failed. Saying that either the pw or name is invalid gives seedy characters a big clue as to which field they now have to crack. I notice the back end also has errors for both pw and email. For the sake of learning to handle different errors and appropriately displaying them this is fine however this is something you generally want to avoid.

Reply:
Got it, I'll fix it.

Reply:
we have error messages that comes from Back end, it is better to catch them.